src/Editor/main.cpp

```diff
@@ -62,53 +52,65 @@ int main() {
         float deltaTime = static_cast<float>(glfwGetTime() - lastTime);
         lastTime = glfwGetTime();
         
+        Managers().profilingManager->SetActive(profiling);
+
         // Begin new profiling frame.
-        Managers().profilingManager->BeginFrame();
+        if (Managers().profilingManager->Active())
+            Managers().profilingManager->BeginFrame();
         
         { PROFILE("Frame");
+        { GPUPROFILE("Frame", Video::Query::Type::TIME_ELAPSED);
```

Comment:
We may want to indent this w.r.t. the previous profile, or put it on the same line.